compiler/hash-layout/src/lib.rs

```diff
@@ -56,18 +62,25 @@ pub struct PointeeInfo {
     pub kind: Option<PointerKind>,
 }
 
-// Define a new key to represent a particular layout.
-new_store_key!(pub LayoutId, derives = Debug);
+stores!(
+    LayoutStores;
+    layouts: LayoutStore
+);
+
+/// The global [`LayoutStores`] instance.
+static STORES: OnceLock<LayoutStores> = OnceLock::new();
+
+/// Access the global [`LayoutStores`] instance.
+pub(crate) fn layout_store() -> &'static LayoutStores {
+    STORES.get_or_init(LayoutStores::new)
+}
```

Comment:
```suggestion
pub(crate) fn layout_store() -> &'static LayoutStore {
    STORES.get_or_init(LayoutStores::new).layouts()
}
```

Reply:
Oh, but you cannot do this since the `static_single_store` requires it to return `LayoutStores` so that it can index into layouts.

Reply:
Oh yeah you're right.. I guess it is fine then. It is just potentially annoying to have to do `layout_store().layouts()` all the time

Reply:
Yeah its only in one place anyway so its not too bad.